### PR TITLE
[encoding] add csv parse

### PR DESCRIPTION
**Pull request by zekth**
Ported https://github.com/zekth/deno_easyCSV to encoding/csv.
Also added the possibility to use `BufReader` as input and not only `string`.

ref: https://github.com/denoland/deno_std/issues/455

### CODE REVIEW

**Comment by ry** [comment]
Cool - looks good - but I'm holding for #444.

**Comment by zekth** [comment]
Rebased.
Passes tests but still have the uTime problem on the MacOs ci

**Comment by ry** [comment]
> uTime problem on the MacOs ci

What's that? Can you submit a PR to disable it until we have a fix?

**Comment by zekth** [comment]
> > uTime problem on the MacOs ci
> 
> What's that? Can you submit a PR to disable it until we have a fix?

here are the related issues:
https://github.com/denoland/deno/issues/2411
https://github.com/denoland/deno_std/issues/421

i'll disable the test so.

File: fs/copy_test.ts [hunk 1]
```diff
@@ -346,8 +346,10 @@ testCopySync(
 
     assert(typeof destStatInfo.accessed === "number");
     assert(typeof destStatInfo.modified === "number");
-    assertEquals(destStatInfo.accessed, srcStatInfo.accessed);
-    assertEquals(destStatInfo.modified, srcStatInfo.modified);
+    // TODO: Activate test when https://github.com/denoland/deno/issues/2411
+    // is fixed
+    // assertEquals(destStatInfo.accessed, srcStatInfo.accessed);
+    // assertEquals(destStatInfo.modified, srcStatInfo.modified);
```
Comment (ry) [inline fs/copy_test.ts]: Can you do this in a separate PR? It seems unrelated to CSV.
